### PR TITLE
Feature/debug renderer double buffer

### PR DESCRIPTION
**Pull request by GlynLeine**

### CODE REVIEW

**Review by Developer-The-Great — APPROVED** [review]

File: args/core/math/glm/detail/type_vec3.hpp [hunk 1]
```diff
@@ -22,6 +22,13 @@ namespace args::core::math
 		typedef vec<3, T, Q> type;
 		typedef vec<3, bool, Q> bool_type;
 
+        static const type up;
+        static const type down;
+        static const type right;
+        static const type left;
+        static const type forward;
+        static const type backward;
+
```
Comment (Algo-ryth-mix) [inline args/core/math/glm/detail/type_vec3.hpp:31]: what is the use of these ? as far as I can tell these should be initialized here!

File: args/core/math/glm/detail/type_vec3.inl [hunk 2]
```diff
@@ -165,26 +178,26 @@ namespace args::core::math
...
-		return y;
-			case 2:
-		return z;
+		case 0:
+			return x;
+		case 1:
+			return y;
+		case 2:
+			return z;
 		}
```
Comment (Algo-ryth-mix) [inline args/core/math/glm/detail/type_vec3.inl:193]: you could also add an operator for char types to allow for this syntax: `v['x']` which would be useful if we want to expose this operator to a scripting language

File: args/core/math/glm/detail/type_vec3.hpp (hunk 1 above)
Comment (GlynLeine) [reply args/core/math/glm/detail/type_vec3.hpp:31]: since they're static they can't be initiliased here

Comment (Algo-ryth-mix) [reply args/core/math/glm/detail/type_vec3.hpp:31]: they are also const

File: args/core/math/glm/detail/type_vec3.inl (hunk 2 above)
Comment (GlynLeine) [reply args/core/math/glm/detail/type_vec3.inl:193]: would be usefull but this is just default glm

File: args/core/math/glm/detail/type_vec3.hpp (hunk 1 above)
Comment (Algo-ryth-mix) [reply args/core/math/glm/detail/type_vec3.hpp:31]: ah well they aren't pod so I guess they can't be static constexpr, which would probably be the ideal case

**Pull request merged**